Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -168,275 +194,229 @@ - (void)viewWillAppear:(BOOL)animated
-                                    forState:UIControlStateNormal];
-    [inviteButton.titleLabel setFont:[UIFont ows_regularFontWithSize:17.f]];
+#pragma mark - Table Contents
 
-    UIButton *searchByPhoneNumberButton = [UIButton buttonWithType:UIButtonTypeCustom];
-    [searchByPhoneNumberButton setTitle:NSLocalizedString(@"NO_CONTACTS_SEARCH_BY_PHONE_NUMBER",
-                                                          @"Label for a button that lets users search for contacts by phone number")
-                               forState:UIControlStateNormal];
-    [searchByPhoneNumberButton setTitleColor:[UIColor ows_materialBlueColor]
-                                    forState:UIControlStateNormal];
-    [searchByPhoneNumberButton.titleLabel setFont:[UIFont ows_regularFontWithSize:17.f]];
-    [inviteButton.superview addSubview:searchByPhoneNumberButton];
-    [searchByPhoneNumberButton autoHCenterInSuperview];
-    [searchByPhoneNumberButton autoPinEdge:ALEdgeTop
-                                    toEdge:ALEdgeBottom
-                                    ofView:inviteButton
-                                withOffset:20];
-    [searchByPhoneNumberButton addTarget:self
-                                  action:@selector(hideBackgroundView)
-                        forControlEvents:UIControlEventTouchUpInside];
-}
-
-- (void)hideBackgroundView {
-    [[Environment preferences] setHasDeclinedNoContactsView:YES];
-    
-    [self showEmptyBackgroundViewIfNecessary];
-}
-
-- (void)presentInviteFlow
+- (void)updateTableContents
 {
-    OWSInviteFlow *inviteFlow =
-        [[OWSInviteFlow alloc] initWithPresentingViewController:self contactsManager:self.contactsManager];
-    [self presentViewController:inviteFlow.actionSheetController animated:YES completion:nil];
-}
+    OWSTableContents *contents = [OWSTableContents new];
 
-- (void)showLoadingBackgroundView:(BOOL)show {
-    if (show) {
-        self.searchController.searchBar.hidden = YES;
-        self.tableView.backgroundView          = _loadingBackgroundView;
-        self.tableView.backgroundView.opaque   = YES;
-    } else {
-        self.searchController.searchBar.hidden = NO;
-        self.tableView.backgroundView          = nil;
+    if (self.isNoContactsModeActive) {
+        self.tableViewController.contents = contents;
+        return;
     }
-}
-
-- (void)showEmptyBackgroundViewIfNecessary {
-    self.isNoContactsViewVisible = ([self.contacts count] == 0 &&
-                                    ![[Environment preferences] hasDeclinedNoContactsView]);
-}
 
-- (void)setIsNoContactsViewVisible:(BOOL)isNoContactsViewVisible {
-    if (isNoContactsViewVisible == _isNoContactsViewVisible) {
-        return;
+    __weak MessageComposeTableViewController *weakSelf = self;
+    ContactsViewHelper *helper = self.contactsViewHelper;
+
+    OWSTableSection *section = [OWSTableSection new];
+
+    // Find Non-Contacts by Phone Number
+    [section addItem:[OWSTableItem itemWithCustomCellBlock:^{
```

Comment:
This is somewhat redundant with the existing search bar, but the "find by phone number" is nice in that it's an easier way to key in the numbers (number pad vs. keyboard) and an easier way to find someone in a different country code.

However, I like the existing search bars better "autocomplete"-like behavior with "Start conversation with #" and the "invite # via sms". The "Find by phone number view" doesn't give any feedback until you "submit".

Also, it's taking up some real prime real estate, especially on smaller devices:

![search_invite](https://cloud.githubusercontent.com/assets/217057/25711367/bea5041c-30bc-11e7-82ec-d859540b9334.PNG)

Another thing, when I type in part of a phone number, e.g. "555-123-12", "Find by phone number" almost seems like a button to execute the search for accounts based on my incomplete search input.

Weighing those things, I'm not sure we should keep it. WDYT @charlesmchen?

In either case, on a separate design note, I believe the top two items probably *are* the same height as the rest of the table cells, but without the avatars, the top two items feel a little too big.  Do you think it would be unreasonable to make them shorter than the other items in the table?